src/main/java/com/saralein/server/server/Server.java

```diff
@@ -42,4 +47,13 @@ public void run() {
     public void stop() {
         listening = false;
     }
+
+    private void setServerIP() {
+        try {
+            InetAddress address = InetAddress.getLocalHost();
+            serverIP = address.getHostAddress();
```

Comment:
Would it be possible to set these via a constructor? Doing the initial setting of an instance variable outside of the field declaration or constructor feels like a code smell to me.

Reply:
(But there are exceptions -- like with the Builder Pattern)